### DISCUSSION TITLE
Get value from field of object

### DISCUSSION THREAD

**Discussion by Lukasz96** [i18next/i18next#2202] [Q&A] [answered]
My translation file is in format
```
 "key1": {
    "text": "some value",
    "description": "desc"
  },
  "key2": {
    "text": "some value",
    "description": "desc"
  }...
```
I want to get `text` field from given key, ex `key1`. When I enable  `returnObjects` to `true` and use `postProcessor` to get wanted field from object response it does not work - seems like `postProcessor` goes through `text` field, then `description` field, and at the end translation return whole object anyway, which results with error in React then. 
Any ideas how to get only wanted field to have it then passed to my React component?

**Comment by jamuhl** [comment]
a) recreate your JSON as i18next need it
b) call `t('key1.text)`

**Reply by Lukasz96** [reply]
Unfortunately I cannot recreate this, as this is new standard for translation files in my company and I have just migrated into this. The option with `t('key1.text)` would work but this would mean refactoring hundreds of files in my app, so I wanted to find way to format the translation result globally by 18next settings - just extract from response object wanted field. In this way I would not have to change anything in business code.

**Reply by jamuhl** [reply, marked as answer]
idk...there is a difference between a translations file for translation management and translations file for i18n libs...but that might be just my opinion (would your company use our locize.com there wouldn't be such problems to start with).

easiest todo...write a small script that transforms the JSON...

if your using the http-backend you can transform on the fly passing a parse function in options: https://github.com/i18next/i18next-http-backend?tab=readme-ov-file#backend-options

**Reply by Lukasz96** [reply]
Hi, I tried with transforming the data on import, it was the best option - no changes in other places, thanks! I just removed the `.text` suffix from every key so that all places in code do not see difference in new json format.
Maybe the code will be helpful for someone in future:
```
import { flatten } from 'flat'
...
.use(
    resourcesToBackend(async (language: string, ns: string) => {
      const translation = await import(`./translations/strings.json`)
      const flattenedTranslation: { [key: string]: string } = flatten(translation)
      Object.keys(flattenedTranslation).forEach((key) => {
        if (key.endsWith('.text')) {
          const newKey = key.substring(0, key.length - 5)
          flattenedTranslation[newKey] = flattenedTranslation[key]
          delete flattenedTranslation[key]
        }
      })
      return flattenedTranslation
    })
)
.init(...)
```
